### DISCUSSION TITLE
The Complete Rewrite Commences!

### DISCUSSION THREAD

**Discussion by KeyPuncher** [KeyPuncher/WiinUPro#183] [Announcements]
It is time. 

I started this project over a decade ago. When the Wii U Pro controller was released a week before the Wii U came out I had one and was already attempting to get it connected to Windows. There has been lots of learnings, helpful resources, shared discoveries and a passionate community along the way. As much as I had been aspiring to create version 1.0.0 of WiinUPro, it seems now is the time to simply accept that the project needs a technical refresh.

I have started rewriting an application to replace WiinUPro. So far (I say that because I'm indecisive), I'm calling it _ShiftPad_. With this new code base I'm using more modern approaches and will make the code much easier to change compared to WiinUPro and hopefully more relievable too. I'm targeting newer frameworks and hope to solve Windows 11 compatibility and beyond. By using portable libraries that are broken into modules I hope to make it easy to extend and add new types of controller support from other and future consoles as well as make other operating system distributions possible.

_ShiftPad_ is still quite a ways out but I'm already fairly satisfied with the progress from just the past week of working on it during nights. If you would like to support this endeavor, post here with your thoughts. Got alternate name ideas? List them. Are you a graphic designer? Throw out some logo ideas. UX designer? Usage mocks or improvements. General user? At the risk of scope creep, through out some ideas, it will help me gauge my own with what the community is interested in.

**Comment by StrScr** [comment]
I personally haven't used this software in a while since I now mostly use controllers with native Windows support, but it was invaluable when the Wii U Pro controller was the only controller I was willing to use on PC. It's nice to see where you are taking this! I like the name ShiftPad. This is just a pipe dream but it would be interesting to see Wii U gamepad support one day. I've seen videos covering how it's being reverse engineered and I think it could become a great PC accessory.

**Comment by Davis8483** [comment]
This would be amazing. I haven't wanted to shell out money for dedicated controllers when I have perfectly working Wii U controllers. The day I found out WinU Pro was broken I spent sooooo long trying to find a workaround but sadly couldn't get it working. If you have a patreon or something similar I would love to become a monthly supporter.

**Reply by Ivanmatthew** [reply]
WiinUPro still works though? I can confirm because I am running it to play black myth wukong.

**Reply by Davis8483** [reply]
are you on windows 10 or 11?

**Reply by Ivanmatthew** [reply]
This is windows 11 I am talking about

**Comment by KILATEGames910** [comment]
Esperando un WiiuSoft que sea compatible como mando de xbox one

**Reply by KeyPuncher** [reply]
Yes! I too would like to make this version support a Xbox One controller style.

**Reply by KeyPuncher** [reply]
I just learned that this might be quite easy since Xbox One and X|S controllers still use XInput, they just have a few differences in the data on the data structure that is being used. I'm probably over looking something though.

**Comment by KeyPuncher** [comment]
Here's an update/non-update. As some of you are aware I'm a grad student & parent & work full time, making this difficult to work on while school is in session but I have had small spats of planning. My current plan is to jump into the design phase of the project mid to late December and I plan to try and be as transparent as possible for the progress. I've had ideas on how to make the application extensible, using plugins for supporting controllers such that you won't have to download a new build of the application to get additional controller support, just add a plugin, and theoretically any developer could write a plugin for additional controller support.

I'm not sure exactly how far I'll get with the design (by design I mean both the logical software design and UI/UX design) before my next class starts in January, thus I may not actually get good progress until Summer 2025. The somewhat good news is that I'll be completely done with school before Summer 2026, so there's that. Still a ways away and my tendency to want to start projects is ever present and will likely get in the way at times, but I'm working on trying to keep focus on less things, we'll see how it goes. Everyone's support has been really great and encouraging too!

**Comment by Rickyneer** [comment]
I will try and convert the project to be using ViGeM

**Comment by KeyPuncher** [comment]
Hello all! As suspected in my previous update post, I have not had the time to continue, however development will continue late May and last throughout the Summer. The goal is to bring the project to being usable and when Switch 2 controllers drop, start working on adding support for those as well with the goal of having a usable and satisfactory application with as much feature support as possible before my next class (second to last!) begins.

Though I've never really liked the idea of asking for funding, I'm considering putting up a small crowd funding campaign for the costs of the Joy Con 2s and Switch 2 Pro Controller for adding support since together that's $170 before taxes😬and on the chance if there happened to be any additional support funds that would go toward giving _Shift-Pad_ a permeate home and taking time off work to work on the project. Since Patreon is geared towards subscriptions and Kickstart has had some vocal criticisms form the community, I'm thinking GoFundMe or IndieGoGo would be the most appropriate. I'm unsure, let me know if any of you have thoughts on this.

I'm itching to get back to coding!
